### PR TITLE
add new archetype: Spring Boot webapp with GWT RPC

### PR DESCRIPTION
**Pull request by FrankHossfeld**
Added a new archetype: modular-springboot-webapp which creates a webapp using Spring Boot instead of Jetty or Tomcat. It shows also how to use GHWT RPC with Spring Boot.

### CODE REVIEW

**Review by tbroyer — COMMENTED** [review]
Thanks Frank, but I won't be merging this, for various reasons:

 * first, my aversion for Spring :man_shrugging: 
 * related to the above, I don't want to maintain such code (and it **will** have to be maintained, from time to time when new SpringBoot versions are released at a minimum –contrary to the others where Jetty/Tomcat are only *auxiliary* dependencies, and Guice doesn't have a lot of releases :wink:–; and there could be questions and bugs)
 * it works slightly differently from the other modules here

I encourage you to put this archetype in your own project and deploy/publish it on your side. I'd be happy to add a link to that project from the README for people looking for a SpringBoot-oriented archetype (ideally, those projects should be linked from gwtproject.org if we think they're worth it)

**Comment by FrankHossfeld** [comment]
> * first, my aversion for Spring 🤷‍♂️

Me too, Spring is something like a deep blue whole that pulls you done. But, many of my clients are using Spring Boot, so It's hard to say no to Spring. And to be honest, Spring Boot as a base for a GWT application is nice. Easy way to get a rest controller working. 

Yep, you are right. It works slightly different, so a separate repo is a good idea. I will move the PR into a new repo, once I'll have added your feedback. A link to the new repo would be perfect. I will link this repo, too, of course.   

But first I will fix the things you mentioned. :-)

And, thanks a lot!

File: modular-springboot-webapp/src/test/resources/projects/basic-webapp/reference/basic-webapp-server/pom.xml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-dependencies</artifactId>
+        <version>${spring-boot.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply modular-springboot-webapp/src/test/resources/projects/basic-webapp/reference/basic-webapp-server/pom.xml:22]: Move it to the root. Also removed javax-servlet-api

File: modular-springboot-webapp/src/test/resources/projects/basic-webapp/reference/basic-webapp-server/pom.xml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          <value>dev</value>
+        </property>
+      </activation>
+      <build>
+        <pluginManagement>
+          <plugins>
+            <plugin>
+              <groupId>org.springframework.boot</groupId>
+              <artifactId>spring-boot-maven-plugin</artifactId>
+              <version>${spring-boot.version}</version>
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply modular-springboot-webapp/src/test/resources/projects/basic-webapp/reference/basic-webapp-server/pom.xml:89]: removed

File: README.md [hunk 3]
```diff
@@ -36,8 +37,8 @@ Change directory to your generated project and issue the following commands:
 
 1. In one terminal window: `mvn gwt:codeserver -pl *-client -am`
 2. In another terminal window: `mvn jetty:run -pl *-server -am -Denv=dev`
-
-Or if you'd rather use Tomcat than Jetty, use `mvn tomcat7:run` instead of `mvn jetty:run`.
+3. 
+Or if you'd rather use Tomcat than Jetty, use `mvn tomcat7:run` instead of `mvn jetty:run`. And if you'd rather use SpringBoot, use `mvn spring-boot:run -pl *-server -am -Denv=dev` instead of `mvn jetty:run`.
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply README.md:41]: will be updated, when moved to it's own repo.

File: modular-springboot-webapp/src/main/resources/archetype-resources/__rootArtifactId__-server/src/main/java/Application.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply modular-springboot-webapp/src/main/resources/archetype-resources/__rootArtifactId__-server/src/main/java/Application.java:22]: Arrrrggggghhhh ... copy & paste bug ...

File: modular-springboot-webapp/src/test/resources/projects/basic-webapp/reference/basic-webapp-server/pom.xml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>archetype.it</groupId>
+    <artifactId>basic-webapp</artifactId>
+    <version>0.1-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>basic-webapp-server</artifactId>
+  <packaging>war</packaging>
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply modular-springboot-webapp/src/test/resources/projects/basic-webapp/reference/basic-webapp-server/pom.xml:12]: It depends, I want a deployable war. Generating a jar will add the stuff for a Tomcat instance.

File: modular-springboot-webapp/src/main/resources/archetype-resources/pom.xml [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>net.ltgt.gwt.maven</groupId>
+        <artifactId>gwt-maven-plugin</artifactId>
+        <inherited>false</inherited>
+        <configuration>
+          <launcherDir>${dollar}{spring-boot.public.dir}</launcherDir>
+          <warDir>${dollar}{spring-boot.public.dir}</warDir>
```
[the remark this answers is not in the document]
Comment (FrankHossfeld) [reply modular-springboot-webapp/src/main/resources/archetype-resources/pom.xml:43]: removed ...

**Comment by FrankHossfeld** [comment]
Do you think `gwt-maven-springboot-architype' is a suitable name for the new repo?

**Comment by FrankHossfeld** [comment]
I'll close the PR, because I have created a new repo and move the PR into this repo - as you suggested. You can find the repo here: https://github.com/NaluKit/gwt-maven-springboot-archetype. 

Thanks for your help

**Pull request closed without merging**